harness/determined/launch/deepspeed.py

```diff
@@ -139,6 +173,16 @@ def main(script: List[str]) -> int:
     # Chief IP is set as an environment variable to support nested launch layers
     os.environ["DET_CHIEF_IP"] = chief_ip
 
+    # If the NCCL_SOCKET_IFNAME environment variable wasn't explicitly set by
+    # the user in the experiment's YAML file, then set it to the distributed
+    # network interface, if the value of "dtrain_network_interface" under
+    # "task_container_defaults" has been set in the "master.yaml".
+    if is_using_cuda() and not is_nccl_socket_ifname_env_var_set():
+        dtrain_network_interface = os.environ["DET_INTER_NODE_NETWORK_INTERFACE"]
```

Comment:
you'll need to do `os.environ.get("DET_INTER_NODE_NETWORK_INTERFACE", None)` otherwise this will raise a `KeyError` when not present